model/roles.go

```diff
@@ -746,6 +752,56 @@ func (r *Role) calculateRoleConfigurationTemplates() {
 	r.Configuration.Templates = roleConfigs
 }
 
+// WriteConfigs merges the job's spec with the opinions and returns the result as JSON.
+func (roleJob *RoleJob) WriteConfigs(role *Role, lightOpinionsPath, darkOpinionsPath string) ([]byte, error) {
+	var config struct {
+		Job struct {
+			Name string `json:"name"`
+		} `json:"job"`
+		Parameters map[string]string      `json:"parameters"`
+		Properties map[string]interface{} `json:"properties"`
+		Networks   struct {
+			Default map[string]string `json:"default"`
+		} `json:"networks"`
+		ExportedProperties []string               `json:"exported_properties"`
+		Consumes           map[string]jobLinkInfo `json:"consumes"`
+	}
+
+	config.Parameters = make(map[string]string)
+	config.Properties = make(map[string]interface{})
+	config.Networks.Default = make(map[string]string)
+	config.ExportedProperties = make([]string, 0)
+	config.Consumes = make(map[string]jobLinkInfo)
+
+	config.Job.Name = role.Name
+
+	for _, consumer := range roleJob.ResolvedConsumers {
+		config.Consumes[consumer.Name] = consumer.jobLinkInfo
+	}
+
+	opinions, err := NewOpinions(lightOpinionsPath, darkOpinionsPath)
+	if err != nil {
+		return nil, err
+	}
+	properties, err := roleJob.Job.GetPropertiesForJob(opinions)
+	if err != nil {
+		return nil, err
+	}
+	config.Properties = properties
+
+	for _, provider := range roleJob.Job.AvailableProviders {
+		config.ExportedProperties = append(config.ExportedProperties, provider.Properties...)
+	}
+
+	// Write out the configuration
+	jobJSON, err := json.MarshalIndent(config, "", "    ") // 4-space indent
+	if err != nil {
+		return nil, err
+	}
+
+	return jobJSON, nil
```

Comment:
Is it really necessary to make sure `jobJSON`  is `nil` when `err` isn't?  Otherwise this could simply be:

```
	// Write out the configuration
	return json.MarshalIndent(config, "", "    ") // 4-space indent
```

Reply:
I guess I just wanted to be consistent. Happy to change it if you really want to.